### PR TITLE
First version of urdf_geometry_parser

### PR DESCRIPTION
**Pull request by vincentrou**

### CODE REVIEW

**Comment by bmagyar** [comment]
I'll get this in then we talk details ;) --edit: almost...

File: package.xml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+<?xml version="1.0"?>
+<package format="2">
+  <name>urdf_geometry_parser</name>
+  <version>0.2.2</version>
```
Comment (bmagyar) [inline package.xml]: could you change this to 0.0.0 please? this package as is haven't yet been released

File: include/urdf_geometry_parser/urdf_geometry_parser.h [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+#include <ros/ros.h>
+
+#include <urdf_parser/urdf_parser.h>
+
+namespace urdf_geometry_parser {
+
+class UrdfUtilities {
+
+public:
+  UrdfUtilities(ros::NodeHandle& root_nh, const std::string &base_link);
```
Comment (bmagyar) [inline include/urdf_geometry_parser/urdf_geometry_parser.h]: This class needs a new name.
ParsedRobotModel?
RobotModel? (too general probably)
GeometryParser?

File: src/urdf_geometry_parser.cpp [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+#include <urdf_geometry_parser/urdf_geometry_parser.h>
+
+#include <tf2/LinearMath/Vector3.h>
+
+/*
+ * \brief Check if the link is modeled as a cylinder
+ * \param link Link
+ * \return true if the link is modeled as a Cylinder; false otherwise
+ */
+static bool isCylinder(const boost::shared_ptr<const urdf::Link>& link)
```
Comment (bmagyar) [inline src/urdf_geometry_parser.cpp]: These functions should be placed within the `urdf_geometry_parser` namespace.

**Review by bmagyar — CHANGES_REQUESTED** [review]
I'm sorry , forgot to press the green button to submit review :(

**Comment by vincentrou** [comment]
I have made the requested modification.

File: CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+find_package(catkin REQUIRED COMPONENTS ${${PROJECT_NAME}_CATKIN_DEPS})
+## Uncomment this if the package has a setup.py. This macro ensures
+## modules and global scripts declared therein get installed
+## See http://ros.org/doc/api/catkin/html/user_guide/setup_dot_py.html
+# catkin_python_setup()
+
+###################################
+## catkin specific configuration ##
+###################################
+## The catkin_package macro generates cmake config files for your package
```
Comment (bmagyar) [inline CMakeLists.txt]: please remove the boilerplate comments from here

File: src/urdf_geometry_parser.cpp [hunk 5]
```diff
@@ -0,0 +1,204 @@
...
+#include <tf2/LinearMath/Vector3.h>
+
+namespace urdf_geometry_parser{
+
+  /*
+   * \brief Check if the link is modeled as a cylinder
+   * \param link Link
+   * \return true if the link is modeled as a Cylinder; false otherwise
+   */
+  static bool isCylinder(const boost::shared_ptr<const urdf::Link>& link)
```
Comment (bmagyar) [inline src/urdf_geometry_parser.cpp:46]: I think it is better to remove the static-ness of these 2 non-member functions and add a declaration of them to the header so they are visible.

**Review by bmagyar — CHANGES_REQUESTED** [review]

File: src/urdf_geometry_parser.cpp (hunk 5 above)
Comment (vincentrou) [reply src/urdf_geometry_parser.cpp:46]: I don't think it is a good idea to have those functions to be visible.
There are just intend to be used in the getJointRadius function (so not outside of this cpp file).

Comment (bmagyar) [reply src/urdf_geometry_parser.cpp:46]: ideally we'd replace these lines in the `diff_drive_controller` with calls to this library too

https://github.com/ros-controls/ros_controllers/blob/kinetic-devel/diff_drive_controller/src/diff_drive_controller.cpp#L63

https://github.com/ros-controls/ros_controllers/blob/kinetic-devel/diff_drive_controller/src/diff_drive_controller.cpp#L90

https://github.com/ros-controls/ros_controllers/blob/kinetic-devel/diff_drive_controller/src/diff_drive_controller.cpp#L133

Comment (vincentrou) [reply src/urdf_geometry_parser.cpp:46]: Yes this is the purpose of this library.
But the call to [getWheelRadius L546](https://github.com/ros-controls/ros_controllers/blob/kinetic-devel/diff_drive_controller/src/diff_drive_controller.cpp#L546) will be replace by the call to [getJointRadius](https://github.com/vincentrou/urdf_geometry_parser/blob/9fd15e7dffe28c118b567081ba35609289cd02e5/src/urdf_geometry_parser.cpp#L165) in this library (which calls getWheelRadius)

File: package.xml [hunk 6]
```diff
@@ -0,0 +1,22 @@
+<?xml version="1.0"?>
+<package format="2">
+  <name>urdf_geometry_parser</name>
+  <version>0.0.0</version>
```
Comment (davetcoleman) [inline package.xml:4]: needs version number

Comment (vincentrou) [reply package.xml:4]: It was a request of @bmagyar to reset the version number to 0.0.0 https://github.com/ros-controls/urdf_geometry_parser/pull/2#discussion_r117959144
The version number will be incremented when the package is released right ?

**Review by bmagyar — APPROVED** [review]

File: src/urdf_geometry_parser.cpp (hunk 5 above)
Comment (bmagyar) [reply src/urdf_geometry_parser.cpp:46]: Ok, let's leave this pending. I think those functions could be useful outside of this lib, we will see later if making them visible is necessary.

File: package.xml (hunk 6 above)
Comment (bmagyar) [reply package.xml:4]: yep

**Pull request merged**